### PR TITLE
Bring agreements database into this repo

### PR DESCRIPTION
**Pull request by rosskarchner**
This takes ~~two~~ one of our older, private django apps, and brings them in to the light of cfgov-refresh, getting us closer to 100% open source.

~~Originally, this included the 'selfregistration' app, but we now presume this will be merged *after* /company-signup/ is moved into wagtail, and thus we will no longer need that app.~~

'selfregistration' was removed in #3610

## Additions

- the 'agreements' app

## Removals

- the corresponding 'optional-private.txt' lines
- the 'selfregistration' app, and a related feature flag

## Changes

- the apps have been moved from the 'optional apps' system, to being hardcoded in INSTALLED_APPS

-

## Todos

-

## Checklist

* [ ] PR has an informative and human-readable title
* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated
* [ ] Reviewers requested with the [Assignee tool](https://help.github.com/articles/assigning-issues-and-pull-requests-to-other-github-users/) :arrow_right:

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -35,13 +35,16 @@ exclude =
...
 
     # Our settings files might need to conform to different readability
     # standards
     cfgov/cfgov/settings/*,
 
     # Now we come to the bits we exclude because they haven't had a
     # proper cleanup yet.
-    cfgov/v1/tests
+    cfgov/v1/tests,
+    cfgov/agreements/features
```
Comment (Scotchester) [inline .flake8]: Trivial: If the format of this file supports it, add a comma here so you don't have to edit this line next time we add something to this list.

Comment (rosskarchner) [reply .flake8]: I'll look into that

**Comment by rosskarchner** [comment]
I've opened a seperate PR for the selfregistration removal: https://github.com/cfpb/cfgov-refresh/pull/3610

This will be re-focused on just the agreements app

**Comment by rosskarchner** [comment]
This is finally ready to merge: the only new code is tests, otherwise this just the code of the current agreements database.

File: .flake8 [hunk 2]
```diff
@@ -35,13 +35,16 @@ exclude =
     cfgov/jobmanager/migrations/*.py,
     cfgov/v1/migrations/*.py,
     cfgov/ask_cfpb/migrations/*.py,
+    cfgov/selfregistration/migrations/*.py,
+    cfgov/agreements/migrations/*.py,
```
Comment (chosak) [inline .flake8]: While you are editing this file, how would you feel about collapsing this list into the wildcard `*/migrations/*.py`, as suggested in @willbarton's excellent template [here](https://github.com/cfpb/development/pull/154/files#diff-540ed609f086a01ee6f84e1a7bba5f6eR16)?

File: cfgov/agreements/templates/agreements/index.html [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+        credit card issuers as of the first week of January 2016, and does not constitute a full set of agreements).
+        Credit card issuers are generally required to post the credit card agreements that they offer to the public on
+        their websites, with limited exceptions.
+        If you are an issuer, email <a href="mailto:CardAgreements@consumerfinance.gov">CardAgreements@consumerfinance.gov</a>
+        for agreement submission instructions.
+    </p>
+
+
+    <p id="dwnld">
+        <a href="https://s3.amazonaws.com/files.consumerfinance.gov/a/assets/Credit_Card_Agreements_2017_Q2.zip">
```
Comment (chosak) [inline cfgov/agreements/templates/agreements/index.html]: It'd be nice if these weren't hardcoded and there wasn't so much duplication. Do we have any template tag that could be used for these? Perhaps it'd be better for a separate PR, but we could have a tag that provided this prefix under `agreements/templatetags`.

File: .flake8 [hunk 4]
```diff
@@ -35,13 +35,16 @@ exclude =
...
 
     # Our settings files might need to conform to different readability
     # standards
     cfgov/cfgov/settings/*,
 
     # Now we come to the bits we exclude because they haven't had a
     # proper cleanup yet.
-    cfgov/v1/tests
+    cfgov/v1/tests,
+    cfgov/agreements/features,
```
Comment (chosak) [inline .flake8]: Does this directory exist?

File: cfgov/agreements/requirements.txt [hunk 5]
```diff
@@ -0,0 +1,2 @@
+tinys3==0.1.11
```
Comment (chosak) [inline cfgov/agreements/requirements.txt]: Both of these requirements are already in `requirements/base.txt` (with a different version for `tinys3`). Can/should this file be removed?

File: cfgov/agreements/management/commands/import_agreements.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    help = "Upload agreements data from new Quarterly Agreement file path"
+
+    def add_arguments(self, parser):
+        parser.add_argument('-p', '--path', action='store', required=True)
+
+    def handle(self, *args, **options):
+        now = datetime.now()
+        suffix = '%s_%s_' % (now.month, now.year)
+
+        uri_hostname = 'http://files.consumerfinance.gov'
```
Comment (chosak) [inline cfgov/agreements/management/commands/import_agreements.py]: See comment below about S3 paths, it'd be nice to avoid this duplication (very reasonable to be in another PR if better).

File: .flake8 (hunk 2 above)
Comment (rosskarchner) [reply .flake8]: That's a good idea

File: cfgov/agreements/requirements.txt (hunk 5 above)
Comment (rosskarchner) [reply cfgov/agreements/requirements.txt]: Yeah, it should be removed

File: cfgov/agreements/templates/agreements/index.html (hunk 3 above)
Comment (rosskarchner) [reply cfgov/agreements/templates/agreements/index.html]: I was trying *not* to revist application code at this point-- that's a good idea for a seperate PR.

File: cfgov/agreements/management/commands/import_agreements.py (hunk 6 above)
Comment (rosskarchner) [reply cfgov/agreements/management/commands/import_agreements.py]: Let's do it in another PR

File: .flake8 (hunk 2 above)
Comment (rosskarchner) [reply .flake8]: done!

File: cfgov/agreements/requirements.txt (hunk 5 above)
Comment (rosskarchner) [reply cfgov/agreements/requirements.txt]: it's been removed

File: .flake8 [hunk 7]
```diff
@@ -29,19 +29,14 @@ exclude =
     # Generated migration files will throw errors. We need to find a way
     # to exclude django-generated migrations while including
     # manually-written migrations.
-    cfgov/core/migrations/*.py,
-    cfgov/data_research/migrations/*.py,
-    cfgov/flags/migrations/*.py,
-    cfgov/jobmanager/migrations/*.py,
-    cfgov/v1/migrations/*.py,
-    cfgov/ask_cfpb/migrations/*.py,
+    */migrations/*.py
```
Comment (chosak) [inline .flake8]: Is there a missing comma here?

Comment (rosskarchner) [reply .flake8]: fixed in 91c4a12

**Comment by rosskarchner** [comment]
Closed and re-opened to force a Travis run
